src/runtime/hexagon_host.cpp

```diff
@@ -147,7 +147,11 @@ WEAK int init_hexagon_runtime(void *user_context) {
     if (!host_lib) {
         host_lib = halide_load_library("libhalide_hexagon_host.dll");
     }
-
+    if (!host_lib) {
+        // This will now cause a more specific error 'halide_error_code_symbol_not_found' down the line.
+        // So, just print this message and continue on instead of returning a generic error here.
+        error(user_context) << "Hexagon: unable to load libhalide_hexagon_host.so";
```

Comment:
(1) Depending on how you have `halide_error()` configured, this may not continue on at all. 
(2) The golden rule of Halide Runtime code is that every call to `error()` or `halide_error()` *must* *also* return a nonzero error code across the runtime API boundary; otherwise, you get different failure modes in different configurations. 

If you always want to continue from here, use `print()` instead of `error()`.

Reply:
Got it.

Reply:
@pranavb-ca